File: binding/src/libqtestsupport_widgets.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qtest-qtoucheventwidgetsequence.html
pub const qtest__qtoucheventwidgetsequence = struct {

    /// New constructs a new QTest::QTouchEventWidgetSequence object.
    pub fn New(param1: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_new(@ptrCast(param1));
    }


    pub fn Press(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Press(@ptrCast(self), touchId, @ptrCast(pt));
    }

    pub fn Move(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Move(@ptrCast(self), touchId, @ptrCast(pt));
    }

    pub fn Release(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Release(@ptrCast(self), touchId, @ptrCast(pt));
    }

    pub fn Stationary(self: ?*anyopaque, touchId: i32) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Stationary(@ptrCast(self), touchId);
    }

    /// Allows for overriding the related default method
    pub fn OnStationary(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QTest__QTouchEventWidgetSequence) void {
        qtc.QTest__QTouchEventWidgetSequence_OnStationary(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseStationary(self: ?*anyopaque, touchId: i32) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_QBaseStationary(@ptrCast(self), touchId);
    }

    pub fn Commit(self: ?*anyopaque, processEvents: bool) bool {
        return qtc.QTest__QTouchEventWidgetSequence_Commit(@ptrCast(self), processEvents);
    }

    /// Allows for overriding the related default method
    pub fn OnCommit(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) bool) void {
        qtc.QTest__QTouchEventWidgetSequence_OnCommit(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCommit(self: ?*anyopaque, processEvents: bool) bool {
        return qtc.QTest__QTouchEventWidgetSequence_QBaseCommit(@ptrCast(self), processEvents);
    }

    pub fn Press3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, widget: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Press3(@ptrCast(self), touchId, @ptrCast(pt), @ptrCast(widget));
    }

    pub fn Move3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, widget: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Move3(@ptrCast(self), touchId, @ptrCast(pt), @ptrCast(widget));
    }

    pub fn Release3(self: ?*anyopaque, touchId: i32, pt: ?*anyopaque, widget: ?*anyopaque) QtC.QTest__QTouchEventWidgetSequence {
        return qtc.QTest__QTouchEventWidgetSequence_Release3(@ptrCast(self), touchId, @ptrCast(pt), @ptrCast(widget));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTest__QTouchEventWidgetSequence_Delete(@ptrCast(self));
    }
};

File: binding/src/libqtestsupport_widgets.cpp
#include <QPoint>
#include <QPointingDevice>
#define WORKAROUND_INNER_CLASS_DEFINITION_QTest__QTouchEventSequence
#define WORKAROUND_INNER_CLASS_DEFINITION_QTest__QTouchEventWidgetSequence
#include <QWidget>
#include <qtestsupport_widgets.h>
#include "libqtestsupport_widgets.h"
#include "libqtestsupport_widgets.hxx"

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_new(const QTest__QTouchEventWidgetSequence* param1) {
	 return new VirtualQTestQTouchEventWidgetSequence(*param1);
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Press(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt) {
	return new QTest::QTouchEventWidgetSequence(self->press(touchId, *pt));
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Move(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt) {
	return new QTest::QTouchEventWidgetSequence(self->move(touchId, *pt));
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Release(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt) {
	return new QTest::QTouchEventWidgetSequence(self->release(touchId, *pt));
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Stationary(QTest__QTouchEventWidgetSequence* self, int touchId) {
	return new QTest::QTouchEventWidgetSequence(self->stationary(touchId));
}

bool QTest__QTouchEventWidgetSequence_Commit(QTest__QTouchEventWidgetSequence* self, bool processEvents) {
	return self->commit(processEvents);
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Press3(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt, QWidget* widget) {
	return new QTest::QTouchEventWidgetSequence(self->press(touchId, *pt, widget));
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Move3(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt, QWidget* widget) {
	return new QTest::QTouchEventWidgetSequence(self->move(touchId, *pt, widget));
}

QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_Release3(QTest__QTouchEventWidgetSequence* self, int touchId, const QPoint* pt, QWidget* widget) {
	return new QTest::QTouchEventWidgetSequence(self->release(touchId, *pt, widget));
}

// Base class handler implementation
QTest__QTouchEventWidgetSequence* QTest__QTouchEventWidgetSequence_QBaseStationary(QTest__QTouchEventWidgetSequence* self, int touchId) {
	auto* vqtest__qtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
	if (vqtest__qtoucheventwidgetsequence && vqtest__qtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
vqtest__qtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Stationary_IsBase(true);
	return new QTest::QTouchEventWidgetSequence(vqtest__qtoucheventwidgetsequence->stationary(touchId));
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventWidgetSequence_OnStationary(QTest__QTouchEventWidgetSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
	if (vqtest__qtoucheventwidgetsequence && vqtest__qtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
vqtest__qtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Stationary_Callback(reinterpret_cast<VirtualQTestQTouchEventWidgetSequence::QTest__QTouchEventWidgetSequence_Stationary_Callback>(slot));
}
}

// Base class handler implementation
bool QTest__QTouchEventWidgetSequence_QBaseCommit(QTest__QTouchEventWidgetSequence* self, bool processEvents) {
	auto* vqtest__qtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
	if (vqtest__qtoucheventwidgetsequence && vqtest__qtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
vqtest__qtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Commit_IsBase(true);
	return vqtest__qtoucheventwidgetsequence->commit(processEvents);
}
}

// Auxiliary method to allow providing re-implementation
void QTest__QTouchEventWidgetSequence_OnCommit(QTest__QTouchEventWidgetSequence* self, intptr_t slot) {
	auto* vqtest__qtoucheventwidgetsequence = dynamic_cast<VirtualQTestQTouchEventWidgetSequence*>(self);
	if (vqtest__qtoucheventwidgetsequence && vqtest__qtoucheventwidgetsequence->isVirtualQTestQTouchEventWidgetSequence) {
vqtest__qtoucheventwidgetsequence->setQTest__QTouchEventWidgetSequence_Commit_Callback(reinterpret_cast<VirtualQTestQTouchEventWidgetSequence::QTest__QTouchEventWidgetSequence_Commit_Callback>(slot));
}
}

void QTest__QTouchEventWidgetSequence_Delete(QTest__QTouchEventWidgetSequence* self) {
    delete self;
}

